### PR TITLE
Add <inheritdoc /> in migrations

### LINKED ISSUE #10695: Add inheritdoc to migration files
It would be nice if `dotnet ef migrations add` would automatically add
```
#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
```
to the beginning of generated file and
```
#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
```
at the end of file.
I mean both (migrations and designer).

For example:
```
#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member

using Microsoft.EntityFrameworkCore.Migrations;
using System;
using System.Collections.Generic;

namespace OpPIS.Web.Hosting.Migrations
{
    public partial class Fix5 : Migration
    {
        protected override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.AddColumn<int>(
                name: "ParentAnchorX",
                table: "VarDesign_Parameter-ImageManipulations",
                nullable: true);
        }

        protected override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropColumn(
                name: "ParentAnchorX",
                table: "VarDesign_Parameter-ImageManipulations");
        }
    }
}

#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
```

### PR DESCRIPTION
**Pull request by roji**
And suppress CA1814 if multidimensional seed data is present.

Closes #10695

### CODE REVIEW

File: src/EFCore.Design/Migrations/Design/CSharpMigrationsGenerator.cs [hunk 1]
```diff
@@ -78,6 +78,14 @@ public override string Language
             .AppendLine()
             .AppendLine("#nullable disable");
 
+        // Suppress "Prefer jagged arrays over multidimensional" when we have a seeding operation with a multidimensional array
+        if (HasMultidimensionalArray(upOperations.Concat(downOperations)))
```
Comment (ajcvickers) [inline src/EFCore.Design/Migrations/Design/CSharpMigrationsGenerator.cs:82]: How is this related to inheritdoc?

Comment (roji) [reply src/EFCore.Design/Migrations/Design/CSharpMigrationsGenerator.cs:82]: It isn't - it's another thing in the same area that's tracked by that same issue (see https://github.com/dotnet/efcore/issues/10695#issuecomment-950693625). Do you prefer I split it out, or think we shouldn't do it?

Comment (ajcvickers) [reply src/EFCore.Design/Migrations/Design/CSharpMigrationsGenerator.cs:82]: I think we should at least discuss. I really hate adding suppressions, especially for overzealous warnings like this. Is this a modern analyzer warning? If so, what is the justification for it? If it's an old FxCop warning then...

Comment (roji) [reply src/EFCore.Design/Migrations/Design/CSharpMigrationsGenerator.cs:82]: [This rule](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1814) is reported by the standard .NET analyzers that started shipping as part of the SDK in .NET 5.0 ([see docs](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview). It's indeed not enabled by default, but anyone enabling e.g. all the "performance" category diagnostics gets it.

I also hate adding suppressions, which is why I went to the extra trouble of adding it only where we actually generate a multidimensional array. I do think it reduces a bit of needless friction for users who enable most of the rules, but if you're against it we can discuss and remove.

**Review by ajcvickers — APPROVED** [review]

File: eng/Versions.props [hunk 2]
```diff
@@ -28,6 +28,6 @@
     <MicrosoftExtensionsLoggingVersion>7.0.0-alpha.1.22066.4</MicrosoftExtensionsLoggingVersion>
   </PropertyGroup>
   <PropertyGroup Label="Other dependencies">
-    <MicrosoftCodeAnalysisVersion>3.7.0</MicrosoftCodeAnalysisVersion>
+    <MicrosoftCodeAnalysisVersion>4.0.1</MicrosoftCodeAnalysisVersion>
```
Comment (bricelam) [inline eng/Versions.props:31]: Can you move this comment to here?

https://github.com/dotnet/efcore/blob/a204b6210e1f14bd29eb1239a53ab88148b7e2b0/src/EFCore.Analyzers/EFCore.Analyzers.csproj#L30-L31

Comment (bricelam) [reply eng/Versions.props:31]: Doh, I was too slow to comment

Comment (roji) [reply eng/Versions.props:31]: Oh sorry... Can slip it into the next thing (either you or me).

Any idea what 4.0.1 restricts us to in terms of VS?

Comment (bricelam) [reply eng/Versions.props:31]: 2022 (probably worth putting https://aka.ms/roslyn-packages in the comment)
